### PR TITLE
javascript tests

### PR DESCRIPTION
**Pull request by ievans**
Initial version of javascript support in the test suite and one javascript test.

### CODE REVIEW

File: javascript/smells/leftover_debugging.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+rules:
+  - id: javascript-alert
+    patterns: 
+        - pattern-either: 
+          - pattern: alert()
+          - pattern: alert($X)
```
Comment (aryx) [inline javascript/smells/leftover_debugging.yaml]: you can use alert(...)
will match 0 or n times.

Comment (aryx) [reply javascript/smells/leftover_debugging.yaml]: Or do you just want to check for call arities of 0 and 1?

Comment (ievans) [reply javascript/smells/leftover_debugging.yaml]: Yup, arities 0 and 1

**Comment by ievans** [comment]
Fixed sgrep in https://github.com/returntocorp/sgrep/pull/80/

**Pull request merged**